### PR TITLE
SOC-1938 - As a user, I want to see URL OG cards added to posts and replies on the discussion post details page and on the post list page

### PR DESCRIPTION
**Pull request by rafalkalinski**
## Links

https://wikia-inc.atlassian.net/browse/SOC-1938

## Description

As a user, I want to see URL OG cards added to posts and replies on the discussion post details page and on the post list page

## Reviewers

@bkoval @kvas-damian

### CODE REVIEW

File: front/main/app/models/discussion/domain/open-graph.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+OpenGraph.reopenClass({
+	/**
+	 * @param {Object} openGraphData
+	 *
+	 * @returns {array}
+	 */
+	create(openGraphData) {
+		if (!openGraphData) {
+			return this._super();
```
Comment (kvas-damian) [inline front/main/app/models/discussion/domain/open-graph.js]: Does it make sense to create empty object when data if empty? What do you think about just returning `null` here?

File: front/main/app/templates/components/discussion-open-graph.hbs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<a class="og-container" href="{{openGraphData.url}}" title="{{openGraphData.title}}">
+	<img class="og-image" src="{{openGraphData.imageUrl}}" alt="{{openGraphData.title}}">
```
Comment (kvas-damian) [inline front/main/app/templates/components/discussion-open-graph.hbs]: Don't we need some condition if image isn't present?

File: front/main/app/styles/component/_discussion-open-graph.scss [hunk 3]
```diff
@@ -0,0 +1,71 @@
+
+// -- theme --
+
+.og-container {
```
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-open-graph.scss]: Those selectors are very wide, it's possible that they can match classnames in other features. What do you think about wrapping them into a class name that relates to discussions only?

File: front/main/app/styles/component/_discussion-open-graph.scss [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	background-color: $discussion-open-graph-background-color;
```
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-open-graph.scss]: Do we use the same colors on dark theme?

File: front/main/app/components/discussion-open-graph.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+	siteName: Ember.computed('openGraphData.domain', 'openGraphData.siteName', function () {
```
Comment (bkoval) [inline front/main/app/components/discussion-open-graph.js]: Do we want to show the domain twice if there's no siteName?

File: front/main/app/models/discussion/domain/open-graph.js (hunk 1 above)
Comment (bkoval) [inline front/main/app/models/discussion/domain/open-graph.js]: I'd rather return an object with default values :)

File: front/main/app/styles/component/_discussion-open-graph.scss [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+.og-site-name {
+	font-size: $type-min;
+	line-height: $line-height-min;
+	margin: 0 18px 22px;
+}
+
+.og-image {
+	flex-grow: 2;
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-open-graph.scss]: Why 2? I don't see any other flexible elements.

File: front/main/app/templates/components/discussion-open-graph.hbs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<a class="og-container" href="{{openGraphData.url}}" title="{{openGraphData.title}}">
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-open-graph.hbs]: Do we need an additional div on top of that? Maybe we could move this to a component definition and remove it from here?

**Comment by bkoval** [comment]
Thanks for unit tests?

File: front/main/app/models/discussion/domain/open-graph.js (hunk 1 above)
Comment (rafalkalinski) [inline front/main/app/models/discussion/domain/open-graph.js]: It creates the object with default values.

File: front/main/app/templates/components/discussion-open-graph.hbs (hunk 2 above)
Comment (rafalkalinski) [inline front/main/app/templates/components/discussion-open-graph.hbs]: SOC-1940 covers that (different OG card views depends on the data from server)

File: front/main/app/models/discussion/domain/open-graph.js (hunk 1 above)
Comment (kvas-damian) [inline front/main/app/models/discussion/domain/open-graph.js]: I don't get it.
Why do we use default values when OpenGraph doesn't exist for the post?

File: front/main/app/styles/component/_discussion-open-graph.scss (hunk 3 above)
Comment (rafalkalinski) [inline front/main/app/styles/component/_discussion-open-graph.scss]: +done

File: front/main/app/styles/component/_discussion-open-graph.scss (hunk 4 above)
Comment (rafalkalinski) [inline front/main/app/styles/component/_discussion-open-graph.scss]: I planned to make the dark theme work as a part of SOC-1940, when there will be the final shape of those styles.

File: front/main/app/components/discussion-open-graph.js (hunk 5 above)
Comment (rafalkalinski) [inline front/main/app/components/discussion-open-graph.js]: So we want to show there the title and the siteName only. If there is no siteName, then we take the domain as a siteName.

File: front/main/app/styles/component/_discussion-open-graph.scss (hunk 6 above)
Comment (rafalkalinski) [inline front/main/app/styles/component/_discussion-open-graph.scss]: removed

File: front/main/app/templates/components/discussion-open-graph.hbs (hunk 7 above)
Comment (rafalkalinski) [inline front/main/app/templates/components/discussion-open-graph.hbs]: Actually we don't need that. It is changed as a part of SOC-1940 - there will be difference in this behavior between post-list (no link) and post-details (link).

File: front/main/app/models/discussion/domain/open-graph.js (hunk 1 above)
Comment (bkoval) [inline front/main/app/models/discussion/domain/open-graph.js]: Yeah, that's true. Doesn't make sense.

**Pull request closed without merging**